Runtime/Highlight/DefaultHighlighter.cs

```diff
@@ -0,0 +1,445 @@
+        /// </summary>
+        /// <remarks>Every highlight requires an ID to avoid duplications.</remarks>
+        public string StartHighlighting(Material highlightMaterial, string highlightID = null)
+        {
+            if (CanObjectBeHighlighted() == false)
+            {
+                return highlightID;
+            }
+            
+            if (string.IsNullOrEmpty(highlightID))
+            {
+                highlightID = $"{Guid.NewGuid()}";
+            }
+            
+            if (activeHighlights.ContainsKey(highlightID) == false)
+            {
+                activeHighlights.Add(highlightID, highlightMaterial);
+                StartCoroutine(Highlight(highlightMaterial, ()=> IsTheLastKey(highlightID)));
+            }
+            
+            return highlightID;
+        }
+        
+        /// <summary>
+        /// Highlights this object with given <paramref name="highlightTexture"/>.
+        /// </summary>
+        /// <remarks>Every highlight requires an ID to avoid duplications.</remarks>
+        public string StartHighlighting(Texture highlightTexture, string highlightID = null)
+        {
+            if (CanObjectBeHighlighted() == false)
+            {
+                return highlightID;
+            }
+            
+            if (string.IsNullOrEmpty(highlightID))
+            {
+                highlightID = $"{Guid.NewGuid()}";
+            }
+            
+            if (activeHighlights.ContainsKey(highlightID) == false)
+            {
+                Material highlightMaterial = NewHighlightMaterial(highlightTexture);
+                activeHighlights.Add(highlightID, highlightMaterial);
+                StartCoroutine(Highlight(highlightMaterial, ()=> IsTheLastKey(highlightID)));
+            }
+            
+            return highlightID;
+        }
+
+        /// <summary>
+        /// Stops a highlight of given <paramref name="highlightID"/>.
+        /// </summary>
+        public void StopHighlighting(string highlightID)
+        {
+            if (activeHighlights.ContainsKey(highlightID))
+            {
+                activeHighlights.Remove(highlightID);
+            }
+        }
+        
+        /// <summary>
+        /// Regenerates the cached renderers.
+        /// </summary>
+        public void ForceRefreshCachedRenderers()
+        {
+            if (IsHighlighting())
+            {
+                return;
+            }
+            
+            ReenableRenderers();
+
+            if (Application.isPlaying && gameObject.isStatic)
+            {
+                return;
+            }
+            
+            ClearCacheRenderers();
+            RefreshCachedRenderers();
+        }
+
+        protected void ClearCacheRenderers()
+        {
+            renderers = default;
+        }
+        
+        private IEnumerator Highlight(Material highlightMaterial, Func<bool> shouldContinueHighlighting, string highlightID = "")
```

Comment:
shouldContinueHighlighting is always the exact same method used, do we have to put it into a parameter?

Reply:
True, this is kind of a legacy code, before the "condition" for keep highlighting was not based on being the last "ID".